Refreshes token before making a request, in case the token is expired.

lpelorosso opened the pull request.

lpelorosso committed:
Refreshes token before making a request, in case the token is expired.

lpelorosso committed:
Search Bar shows used query when coming back from user detail page.

lpelorosso committed:
Reset was not clearing the search bar query.

lpelorosso committed:
Added userField to disable email verification option on the user edit page.

mostekcm commented on client/components/Users/UserActions.jsx:
```diff
@@ -149,6 +149,10 @@ export default class UserActions extends Component {
     }
 
+    /* Check if resending verification email option is enabled */
+    const falseTriggerEmailVerified = _.filter(this.props.userFields, field => field.property === 'email_verified' && field.allow === false);
```
instead of field.allow, I think we just want field.edit

mostekcm replied:
would be more consistent with the rest of the userFields

mostekcm commented on client/reducers/settings.js:
```diff
@@ -73,6 +73,7 @@ export const settings = createReducer(fromJS(initialState), { // eslint-disable-
         parseFieldSection(field.property, field.edit, 'userField.edit', field.display);
         parseFieldSection(field.property, field.create, 'userField.create', field.display);
         parseFieldSection(field.property, field.search, 'userField.search', field.display);
+        parseFieldSection(field.property, field.allow, 'userField.allow', field.display);
```
won't need this if you stick with field.edit

mostekcm commented on client/actions/auth.js:
```diff
@@ -60,37 +60,120 @@ export function logout(logoutUrl) {
+      // get token from storage
+      const token = sessionStorage.getItem('delegated-admin:apiToken');
+      // if there is no token, or it is expired, try to get one
+      if(!token || isExpired(jwtDecode(token))){
```
It might be nice to cache the expiresAt time so we aren't having to decode the token.  Also, when we "fix" this extension so it uses access tokens instead, we shouldn't be decoding the access token.

mostekcm commented on client/actions/auth.js:
```diff
@@ -60,37 +60,120 @@ export function logout(logoutUrl) {
+    // creates an interceptor to refresh token if needed
+    axios.interceptors.request.use((config) => {
+      // get token from storage
+      const token = sessionStorage.getItem('delegated-admin:apiToken');
```
same comment about storing the expiresAt time

mostekcm commented on client/actions/auth.js:
```diff
@@ -60,37 +60,120 @@ export function logout(logoutUrl) {
+    axios.defaults.headers.common.Authorization = `Bearer ${apiToken}`;
+    axios.defaults.headers.common['dae-locale'] = window.config.LOCALE || 'en';
+
+    sessionStorage.setItem('delegated-admin:apiToken', apiToken);
```
we should be storing the expiresAt time here "instead" of the apiToken.  We shouldn't be storing the API token in local or sessionStorage.  Too easy for someone to nab it from there.  I know this was already doing that, but we might want to just fix this now.

mostekcm commented on client/actions/auth.js:
```diff
@@ -60,37 +60,120 @@ export function logout(logoutUrl) {
+    // invoke check session to get a new token
+    webAuth.checkSession({},
+      function(err, result) {
+        if (err) { // there was an error
```
we should be checking for a login_required error here and redirecting to authorize if we get one.

lpelorosso committed:
- userField.allow is not longer used, .edit is used instead
- token is not longer stored in session, exp claim is stored instead
- if checkSession fails, user is redirected to authorize.

mostekcm commented on client/actions/auth.js:
```diff
@@ -178,8 +171,8 @@ const processTokens = (dispatch, apiToken, returnTo) => {
 export function loadCredentials() {
   return (dispatch) => {
-    const token = sessionStorage.getItem('delegated-admin:apiToken');
-    if (token || window.location.hash) {
```
I can take care of this during a merge, but there are now two checks for window.location.hash.  I'm removing the outer check
